src/Channels.Networking.TLS/SecurityContext.cs

```diff
@@ -0,0 +1,198 @@
+            }
+            try
+            {
+                //Load the available security packages and look for the Unified pack from MS that supplies TLS support
+                if (InteropSspi.EnumerateSecurityPackagesW(out numberOfPackages, out secPointer) != 0)
+                {
+                    throw new InvalidOperationException("Unable to enumerate security packages");
+                }
+                var size = sizeof(SecPkgInfo);
+                for (int i = 0; i < numberOfPackages; i++)
+                {
+                    var package = secPointer[i];
+                    var name = Marshal.PtrToStringUni(package.Name);
+                    if (name == SecurityPackage)
+                    {
+                        _maxTokenSize = package.cbMaxToken;
+
+                        //The correct security package is available
+                        _initOkay = true;
+                        GetCredentials();
+                        return;
+                    }
+                }
+                throw new InvalidOperationException($"Unable to find the security package named {SecurityPackage}");
+            }
+            finally
+            {
+                if (secPointer != null)
+                {
+                    InteropSspi.FreeContextBuffer((IntPtr)secPointer);
+                }
+            }
+        }
+
+        private unsafe void GetCredentials()
+        {
+            CredentialUse direction;
+            CredentialFlags flags;
+            if (_isServer)
+            {
+                direction = CredentialUse.Inbound;
+                flags = CredentialFlags.UseStrongCrypto | CredentialFlags.SendAuxRecord;
+            }
+            else
+            {
+                direction = CredentialUse.Outbound;
+                flags = CredentialFlags.ValidateManual | CredentialFlags.NoDefaultCred | CredentialFlags.SendAuxRecord | CredentialFlags.UseStrongCrypto;
+            }
+
+            var creds = new SecureCredential()
+            {
+                rootStore = IntPtr.Zero,
+                phMappers = IntPtr.Zero,
+                palgSupportedAlgs = IntPtr.Zero,
+                cMappers = 0,
+                cSupportedAlgs = 0,
+                dwSessionLifespan = 0,
+                reserved = 0,
+                dwMinimumCipherStrength = 0, //this is required to force encryption
+                dwMaximumCipherStrength = 0,
+                version = SecureCredential.CurrentVersion,
+                dwFlags = flags,
+                certContextArray = IntPtr.Zero,
+                cCreds = 0
+            };
+
+            IntPtr certPointer;
+            if (_isServer)
+            {
+                creds.grbitEnabledProtocols = InteropSspi.ServerProtocolMask;
+                certPointer = _serverCertificate.Handle;
+                //pointer to the pointer
+                IntPtr certPointerPointer = new IntPtr(&certPointer);
```

Comment:
certPointer is a void*, a pointer to the certificate. You then need a certpointer[] which I fake by doing a pointer to the pointer and setting the array length to 1